### PR TITLE
add bookings controller

### PR DESCRIPTION
**Pull request by thebatu**

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class BookingsController < ApplicationController
+  before_action :set_bookings, only: [:show, :destroy]
```
Comment (peyop) [inline app/controllers/bookings_controller.rb]: c'est `set_booking`, au singulier

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  private
+
+  def bookings_params
+  end
+
+  def set_bookings
+    #optimized query
+    # @bookings = Cocktail.includes(doses: :ingredient).find(params[:id])
```
Comment (Zank94) [inline app/controllers/bookings_controller.rb]: Can remove this method as well as before_action

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  def destroy
+  end
+
+  def show
+  end
+
+  private
+
+  def bookings_params
```
Comment (peyop) [inline app/controllers/bookings_controller.rb]: To be implemented ; name is `booking_params` (`booking` :singular)

File: app/controllers/bookings_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  before_action :authenticate_user!, only: [:create]
+
+  def index
+  end
+
+  def new
+  end
+
+  def create
+    cat = Cat.find(params["cat_id"])
```
Comment (Zank94) [inline app/controllers/bookings_controller.rb]: Can use :cat_id sym instead of string

File: app/controllers/bookings_controller.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    Booking.create(user_id: current_user.id, cat_id: cat.id)
```
Comment (peyop) [inline app/controllers/bookings_controller.rb]: `@booking` must be instanciated with `@booking = Booking.new(booking_params)`
And it needs a check : `if @booking.save ...`

File: app/views/cats/show.html.erb [hunk 6]
```diff
@@ -2,3 +2,17 @@
   <h1>cats#show</h1>
   <%= @cat.name %>
 </div>
+
+
+
+<%= simple_form_for [@cat, @bookings] do |f| %>
+  <% if @cat.errors.any? %>
```
Comment (Zank94) [inline app/views/cats/show.html.erb]: Not sure about this being relevant since we're already in cat's show

Comment (thebatu) [inline app/views/cats/show.html.erb]: error detection is always welcomed

**Pull request merged**